### ISSUE TITLE
Pole rotation bug

### ISSUE THREAD

**Issue by DamienIrving** [SciTools/iris#1696]
I'm wanting to use `iris.analysis.cartography.rotate_pole` for some analysis I'm doing, so I performed a quick test that can be seen at this IPython notebook:
https://github.com/DamienIrving/climate-analysis/blob/master/development/iris_pole_rotation.ipynb

The test simply takes the equator and plots what its coordinates would be in the rotated world. The test seems to perform as expected for different values of pole latitude, however varying the pole longitude does not move the resulting coordinates east/west. Is this a bug, or am I misunderstanding what `rotate_pole` does?

**Comment by rhattersley** [comment]
> varying the pole longitude does not move the resulting coordinates east/west

Try limiting your plot to the first N points (e.g. 100) ... and you'll see that the *order* of values is different. Plotting all the lons and lats against each other is hiding this. That's why if you extend the vertical range of your plots (`plt.ylim(-50, 70)`) you'll see there is a line connecting the highest latitude values.

**Comment by DamienIrving** [comment]
@rhattersley I agree that the order is different, but the line should still move east/west with the location of the pole. To be more explicit, I've updated my example to compare the output of `iris.analysis.cartography.rotate_pole` to the expected output:
https://github.com/DamienIrving/climate-analysis/blob/master/development/iris_pole_rotation.ipynb

**Comment by rhattersley** [comment]
> I agree that the order is different, but the line should still move east/west with the location of the pole.

Agreed. Sorry - I should have spotted that. :unamused: I'll try to take a look later.

**Comment by rhattersley** [comment]
> <del>Agreed.</del>

The problem with your `rotate_equator` function (in its [current form](https://github.com/DamienIrving/climate-analysis/blob/8364a15e1374e0578228ffd33660b9d887c04c5b/development/iris_pole_rotation.ipynb)) is that it's using `rotate_pole` to convert normal lon/lat coordinates into rotated lon/lat coordinates, but then it's plotting those rotated coordinates as if they were normal lon/lat coordinates.

In other words, the line:
```
plt.plot(new_lons, new_lats, transform=proj)
```
is incorrect. It should read:
```
plt.plot(new_lons, new_lats, transform=ccrs.RotatedGeodetic(pole_lon, pole_lat))
```

**Comment by rhattersley** [comment]
Alternatively, if you want to plot `new_lons` and `new_lats` in their native coordinate system (i.e. the rotated pole coordinate system) you would do:
```
plt.axes(projection=ccrs.RotatedPole(pole_lon, pole_lat))
plt.plot(new_lons, new_lats)
```

That's is what I've done in this animation, only I've used `plt.scatter` instead of `plt.plot` to avoid the spurious connecting lines.

![rotated](https://cloud.githubusercontent.com/assets/917914/8084256/00bd8562-0f82-11e5-9f72-628153f36c30.gif)

NB. The green scatter points are the results from `iris.analysis.cartography.rotate_pole`, the black line at the real-world equator is given by:
```
plt.plot(orig_lons, orig_lats, transform=ccrs.Geodetic())
```

**Comment by pelson** [comment]
I like watching the world go around like that. :+1:

**Comment by DamienIrving** [comment]
@rhattersley Thanks! This makes much more sense now.

**Comment by DamienIrving** [comment]
@rhattersley Related to this pole rotation issue, I've posted [this question](https://groups.google.com/forum/#!topic/scitools-iris/c648LB7hYnE) on the google group. Any ideas?

**Comment by pelson** [comment]
Ping @pp-mo - this looks like it might be best answered by you.

**Comment by rhattersley** [comment]
Looks like this issue is resolved.
